frb_codegen/src/source_graph.rs

```diff
@@ -31,7 +34,7 @@ pub struct Crate {
 }
 
 impl Crate {
-    pub fn new(manifest_path: &str) -> Self {
+    pub fn pre_new(manifest_path: &str) -> (String, PathBuf) {
```

Comment:
Hmm the several new looks a bit confusing: pre_new vs new vs new_without_resolve. They even looks like doing different things :/